src/PIL/Image.py

```diff
@@ -2204,7 +2201,7 @@ def save(self, fp, format=None, **params):
 
         preinit()
 
-        ext = os.path.splitext(filename)[1].lower()
+        ext = os.fsdecode(os.path.splitext(filename)[1].lower())
```

Comment:
… the extension format lookup to fail here (b/c all dict items are strings), thereby raising `ValueError`.

(Fix is to ensure that the extension is always a string as above.)